src/soot/Scene.java

```diff
@@ -686,7 +686,7 @@ private void modifyHierarchy() {
      * as a library class and invalidates the class hierarchy.
      * @param c The class to add
      */
-    public void addClass(SootClass c) 
+    public synchronized void addClass(SootClass c) 
```

Comment:
Do you remember why it was necessary to make these method synchronized? As far as I know, class-resolution is serial in Soot and, thus, this wasn't necessary before. I am confused why your PR should change this.

Reply:
Hmm... I certainly saw multiple threads running my code (without specifying an option, which I forget the name of, to limit it to one thread). It is possible something has changed since I originally did this (which was quite a while ago).

Reply:
I think this may be unnecessary now, at least my test case which triggered the problem before doesn't with current HEAD, and my changes. It is possible that some other bug was fixed it the mean time. We'll try it without this change.